### DISCUSSION TITLE
Setting proxy host and port using JDBC properties

### DISCUSSION THREAD

**Discussion by daksha121** [aws/amazon-redshift-jdbc-driver#99] [Q&A]
Is there a way to pass the proxyHost and proxyPort settings to the JDBC driver? I see something supported for ODBC driver [here](https://docs.aws.amazon.com/redshift/latest/mgmt/odbc20-configuration-options.html), but couldn't find it for JDBC. I see that the current implementation reads these properties from environment variables but we have other HTTP traffic that we don't want to use the proxy for.

**Comment by daksha121** [comment]
@bhvkshah appreciate your input on this!

**Comment by bhvkshah** [comment]
Hi @daksha121 . You are correct - the proxyHost and proxyPort settings are fetched from environment variables. The JDBC driver does not currently support setting these via connection properties. 

However, you can also add the following parameters to your client application JVM (Java virtual machine) options:

-Dhttp.useProxy=true
-Dhttps.proxyHost=<proxy_host>
-Dhttp.proxyHost=<proxy_host>
-Dhttps.proxyPort=<proxy_port>
-Dhttp.proxyPort=<proxy_port>
-Dhttp.nonProxyHosts=<list of non proxy host separated by | character>

These are used for authentication via IAM connection (STS and Redshift Coral API uses https) and IDP auth.

**Reply by daksha121** [reply]
Thanks for the response @bhvkshah!

Couple of questions/comments:
1. We have other HTTP traffic on the same machine that we don't want to route via proxy. Just the traffic via JDBC driver. Hence the ask for setting this at the driver level only.
2. If we use username/password based auth, will there be any HTTPS traffic? Ot is this just for IAM based auth?

**Reply by bhvkshah** [reply]
Of course, thanks for bringing this up!

1/ Yes, but is your other HTTP traffic also via the same client application? If not, you can set the variables on your client application's JVM options. Currently there isn't a way to set it at the driver level.
2/ This is for IAM auth as well as for plugin based auth. If ssl is turned on the traffic will be over HTTPS, even for username/password auth.

**Reply by daksha121** [reply]
1) Yes, from the same client application - we will have traffic that we don't want to route via proxy..

**Reply by bhvkshah** [reply]
We can definitely look into adding support for specifying the proxyHost and proxyPort as part of jdbc connection properties, in our future releases. Is this something that would work for you? If so, could you please open an Issue for this?
